components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/kafka/KAFKAMessageListener.java

```diff
@@ -0,0 +1,156 @@
+
+        } catch (Exception e) {
+            logger.error(" Error in Creating Kafka Consumer Connector "
+
+                    + e.getMessage());
+            throw new SynapseException(" Error in Creating Kafka Consumer Connector ",
+                    e);
+        }
+        return isCreated;
+    }
+
+    /**
+     * Starts topics consuming the messages,the message can be consumed by topic or topic filter which are white list and black list.
+     */
+    public void start() throws Exception {
+
+        logger.debug("Start to consume the streams");
+        try {
+            logger.info("Starting KAFKA consumer listener...");
+            Map<String, Integer> topicCount = new HashMap<String, Integer>();
+
+            if (topics != null && topics.size() > 0) {
+                // Define threadCount thread/s for topic
+                for (String topic : topics) {
+                    topicCount.put(topic, threadCount);
+                }
+                Map<String, List<KafkaStream<byte[], byte[]>>> consumerStreams = consumerConnector
+                        .createMessageStreams(topicCount);
+                for (String topic : topics) {
+                    List<KafkaStream<byte[], byte[]>> streams = consumerStreams
+                            .get(topic);
+                    startConsumers(streams);
+
+                }
+            } else if (kafkaProperties.getProperty(KAFKAConstants.TOPIC_FILTER) != null) {
+                // Define #threadCount thread/s for topic filter
+                List<KafkaStream<byte[], byte[]>> consumerStreams;
+                boolean isFromWhiteList = (kafkaProperties
+                        .getProperty(KAFKAConstants.FILTER_FROM_WHITE_LIST) == null || kafkaProperties
+                        .getProperty(KAFKAConstants.FILTER_FROM_WHITE_LIST)
+                        .isEmpty()) ? Boolean.TRUE
+                        : Boolean
+                        .parseBoolean(kafkaProperties
+                                .getProperty(KAFKAConstants.FILTER_FROM_WHITE_LIST));
+                if (isFromWhiteList) {
+                    consumerStreams = consumerConnector
+                            .createMessageStreamsByFilter(
+                                    new Whitelist(
+                                            kafkaProperties
+                                                    .getProperty(KAFKAConstants.TOPIC_FILTER)),
+                                    threadCount);
+                } else {
+                    consumerStreams = consumerConnector
+                            .createMessageStreamsByFilter(
+                                    new Blacklist(
+                                            kafkaProperties
+                                                    .getProperty(KAFKAConstants.TOPIC_FILTER)),
+                                    threadCount);
+                }
+
+                startConsumers(consumerStreams);
+            }
+
+        } catch (Exception e) {
```

Comment:
Why have we caught generic exceptions? If this is done intentionally, adding a comment explaining the reason would be helpful. However, I personally prefer to go for the "|" notation and explicitly specify the list of exceptions caught in the catch block so that, if one of those exceptions (behaviour/other attributes) is changed at a later point, you will get a compile time error indicating how the change needs to be handled.